### PR TITLE
chore: Use mkwinsyscall in lib/auth/webauthnwin

### LINKED ISSUE #31559: webauthnwin: generate types from windows webauthn header
Currently types used to call webauthn.dll are created manually. 
That's approach is quite fragile and can be source of bugs, example: https://github.com/gravitational/teleport/pull/31407

It should be possible to generate types based on header file.

Possible options:
1. https://github.com/xlab/c-for-go 
2. go generate with golang.org/x/sys/windows/mkwinsyscall (windowsauth already uses it in teleport.e).

### PR DESCRIPTION
**Pull request by codingllama**
Use mkwinsyscall to generate the Windows boilerplate in lib/auth/webauthnwin.

Closes #31559.

### CODE REVIEW

**Comment by codingllama** [comment]
CC Zac and Noah as the resident Windows experts.

**Comment by codingllama** [comment]
This took a lot more effort than it looks :/

Tested on Windows using `tsh webauthnwin diag`, which does a register+assert combo. Signatures match and generated code seems to be almost identical to what we had before.

File: lib/auth/webauthnwin/syscall_windows.go
Comment (strideynet) [inline lib/auth/webauthnwin/syscall_windows.go:1]: I don't think it particularly matters - but I did notice a trend of calling these generated files `zsyscall_windows`.

**Review by strideynet — APPROVED** [review]
Much tidier !

File: lib/auth/webauthnwin/syscall_windows.go
Comment (codingllama) [reply lib/auth/webauthnwin/syscall_windows.go:1]: Sure, we can do that. I just didn't wanted the _gen file to be on a Windows-only tag, as I was generating the code in a WSL machine.

Comment (codingllama) [reply lib/auth/webauthnwin/syscall_windows.go:1]: Done.

**Review by tigrato — APPROVED** [review]

**Review by ravicious — APPROVED** [review]
I compiled it on my Windows VM and ran `tsh webauthnwin diag` as well, using my regular key through Parallels.

**Comment by codingllama** [comment]
Thanks for trying it out, Rafał!

File: lib/auth/webauthnwin/webauthn_windows.go [hunk 1]
```diff
@@ -204,50 +172,15 @@ func (n *nativeImpl) MakeCredential(origin string, in *makeCredentialRequest) (*
...
-	return nil
-}
-
 // checkIfDLLExistsAndGetAPIVersionNumber checks if dll exists and tries to load
 // it's version via API call. This function makes sure to not panic if dll is
 // missing.
 func checkIfDLLExistsAndGetAPIVersionNumber() (int, error) {
-	if err := modWebAuthn.Load(); err != nil {
-		return 0, err
-	}
```
Comment (codingllama) [inline lib/auth/webauthnwin/webauthn_windows.go:233]: Note: removing the explicit load was the cause of https://github.com/gravitational/teleport/issues/36851.

**Pull request merged**